atomicfu/src/concurrentTest/kotlin/kotlinx/atomicfu/locks/ThreadParkerTest.kt

```diff
@@ -0,0 +1,39 @@
+package kotlinx.atomicfu.locks
+
+import kotlin.test.Test
+import kotlin.test.assertTrue
+import kotlin.time.Duration
+import kotlin.time.measureTime
+
+class ThreadParkerTest {
+
+    @Test
+    fun parkUnpark() {
+        val mainThread = ParkingSupport.currentThreadHandle()
+
+        testThread {
+            sleepMills(1000)
+            ParkingSupport.unpark(mainThread)
+        }
+
+        val time = measureTime {
+            ParkingSupport.park(Duration.INFINITE)
+        }
+
+        assertTrue(time.inWholeMilliseconds > 900)
```

Comment:
A more reliable (and still less time-consuming) way to test this would be to spin for several thousands of times and see if `park` ever finishes earlier than the corresponding `unpark` (basically, that happens-before is established between the code before `unpark` and the code after `park`).

Also, can't this test be flaky due to spurious wakeups?